src/pages/restaurant/components/dishItems/DishItems.js

```diff
@@ -1,32 +1,106 @@
 // react
-import React, { useMemo } from "react";
+import React, { useMemo, useState } from "react";
 
 //prop-types
 import PropTypes from "prop-types";
 
 //lodash
 import _keys from "lodash/keys";
+import _isEmpty from "lodash/isEmpty";
+
+// helpers
+import {
+  filterItemIfQuantityIsZero,
+  findItemById,
+  updateItemQuantity,
+} from "./helpers/dishItems.helpers";
+import { saveCartItems } from "../../../../utility/localStorage";
 
 // components
 import Sidebar from "../../../../commonComponents/sidebar";
 import Menu from "./components/menu";
+import IconWithLabel from "../../../../commonComponents/iconWithLabel";
+import Cart from "../../../../commonComponents/cart";
+
+// icons
+import cartEmpty from "../../../../assets/images/cart_empty.png";
 
 // css
 import "./dishItems.css";
 
-const DishItems = (props) => {
-  const { dishItems } = props;
+const DishItems = ({ dishItems }) => {
+  const [cart, setCart] = useState({
+    items: [],
+  });
+
+  const onCartChange = (id, newQuantity = 0) => {
+    const { items } = cart;
+
+    const itemFound = findItemById(items, id);
+
+    let updatedCart;
+    if (!itemFound) {
+      updatedCart = [
+        ...items,
+        {
+          productId: id,
+          quantity: newQuantity,
+        },
+      ];
+    } else {
+      updatedCart = updateItemQuantity(items, id, newQuantity);
+    }
+
+    const filteredCart = filterItemIfQuantityIsZero(updatedCart);
+
+    setCart({ items: filteredCart });
+  };
+
+  const onCheckout = () => {
+    saveCartItems(cart);
+    setCart({ items: [] });
+  };
+
+  const renderCartItems = () => {
+    if (_isEmpty(cart) || _isEmpty(cart?.items)) {
+      return (
+        <div className="cart-empty">
+          <h3>Cart Empty</h3>
+          <IconWithLabel
+            className="cart-empty-icon-label"
+            icon={cartEmpty}
+            label="Good food is always cooking! Go ahead, order some yummy items from the
+          menu"
+          />
+        </div>
+      );
+    }
+    return (
+      <Cart
+        cart={cart}
+        products={dishItems}
```

Comment:
although this is correct but you can send it as dishItems which will help in understanding the code and being consistent with naming

Reply:
actually to make the cart component reusable I didn't name the prop as dishItems